test/e2e/branding_test.go

```diff
@@ -0,0 +1,198 @@
+func setupBrandingTestCase(t *testing.T) (*testframework.Clientset, operatorsv1.Brand, *v1.ConfigMap) {
+	client := testframework.MustNewClientset(t, nil)
+	// Get the original operator config
+	originalConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	originalConfigBrand := originalConfig.Spec.Customization.Brand
+	// Get the original Managed Config Map
+	originalManagedConfigMap, err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Get(consoleapi.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		t.Fatalf("could not get console-config configmap, %v", err)
+	}
+
+	return client, originalConfigBrand, originalManagedConfigMap
+}
+
+func cleanupBrandingTestCase(t *testing.T, client *testframework.Clientset, originalConfigBrand operatorsv1.Brand, originalManagedConfigMap *v1.ConfigMap) {
+
+	time.Sleep(1)
+
+	setOperatorConfigBrand(t, client, originalConfigBrand)
+
+	if originalManagedConfigMap.Name != "" {
+		originalManagedConfigMap.SetResourceVersion("0")
+	} else {
+		originalManagedConfigMap = createConsoleConfigMap(operatorsv1.BrandOKD)
+	}
+	err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Delete(consoleapi.OpenShiftConsoleConfigMapName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not delete managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+	_, err = client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Create(originalManagedConfigMap)
+	if err != nil {
+		t.Fatalf("could not reset managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+
+}
+
+// Test setting brand using the operator config which overrides the default and managed configmap
+func TestOperatorConfigBranding(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set a temporary managed config to test it does not override the operator config values
+	_, err := updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, createConsoleConfigMap(operatorsv1.BrandOKD))
+	if err != nil {
+		t.Fatalf("error: could not apply managed config map %v", err)
+	}
+
+	for _, expectedBrand := range validBrands {
+		t.Logf("update operator with %v", expectedBrand)
+		// helper to update the operator config
+		setOperatorConfigBrand(t, client, expectedBrand)
+		// now check if it has set the brand
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			//gotBrand := getOperaterConfigBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+
+}
+
+// Test setting brand via the config map in the openshift-config-managed namespace, this requires the operator config not be set
```

Comment:
```golang
// this e2e test validates that 
// - if console-config in openshift-config-managed does not exist, branding is okd 
// - if console-config in openshift-config-managed does exist, branding is ocp
func TestBrandingFromManagedConfigMap(t *testing.T) {
	// create client 
	// delete managed config, start from default state of no managed config 
	// check console-config in openshift-console, see if brand is ocp 
	// throw error if incorrect
	// create managed config 
	// check console-config in openshift-console, see if brand is okd 
	// throw error if incorrect
}
```

Reply:
I worry about this test (default is `ocp` when nothing else is set).  The brand is only `ocp` when built as a container with the dockerfile.  If someone builds this manually without the `ADDITIONAL_GOTAGS="ocp"` tags, then the default is actually `okd` and a test like this would fail.  E2E test designed around people building the code with specific tags seems like a bad idea.

Reply:
Thinking about `ADDITIONAL_GOTAGS="ocp"`.
Honestly trying to remember if we can deprecate that, since we added the `managed configmap`.  I don't know why we want yet another layer of `brand` fiddling.

Reply:
(Meaning I had thought that `ADDITIONAL_GOTAGS="ocp"` was the old way, and we decided against that because the `managed` configmap allows other components, such as the oauth client, to also retrieve the brand, as a canonical source of truth. )

Reply:
If the `GOTAGS` are deprecated, we should remove that from the Dockerfile in a different PR.  That would make the default `okd` and then the managed config would set it to `ocp`. That work for you guys @benjaminapetersen  @spadgett ?

Reply:
Ok thats fine. We can discuss removing `GOTAGS` as a follow.  Wanted to make sure we are testing the right things.